### PR TITLE
tidb: Check parser S/R conflict in Makefile

### PR DESCRIPTION
**Pull request by shenli**

### CODE REVIEW

**Comment by ngaut** [comment]
LGTM

File: Makefile [hunk 1]
```diff
@@ -30,7 +30,8 @@ parser:
 	a=`mktemp temp.XXXXXX`; \
 	goyacc -o /dev/null -xegen $$a parser/parser.y; \
 	goyacc -o parser/parser.go -xe $$a parser/parser.y; \
-	rm -f $$a; \
+	rm -f $$a;
+	@grep " conflict on " y.output | awk '{print} END {if (NR > 0) {print "Find " NR " conflict in parser.y. Please check y.output for more information."; exit 1}}'
```
Comment (qiuyesuifeng) [inline Makefile]: How about using printf?
I think printf is much clearer.

Comment (siddontang) [inline Makefile]: checking the goyacc output like `2>&1 | grep "conflicts"` for above command may be more fast than grep whole output file.

Comment (shenli) [inline Makefile]: PTAL

**Pull request merged**